x/compute/client/cli/query.go

```diff
@@ -195,6 +220,44 @@ func GetCmdGetContractInfo(cdc *codec.Codec) *cobra.Command {
 	}
 }
 
+func CmdDecryptText(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "decrypt [encrypted_data]",
+		Short: "Attempt to decrypt an encrypted blob",
+		Long: "Attempt to decrypt a base-64 encoded encrypted message. This is intended to be used if manual decrypt" +
+			"is required for data that is unavailable to be decrypted using the 'query compute tx' command",
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			encodedInput := args[0]
+
+			dataCipherBz, err := base64.StdEncoding.DecodeString(encodedInput)
+			if err != nil {
+				return fmt.Errorf("error while trying to decode the encrypted output data from base64: %w", err)
+			}
+
+			nonce := dataCipherBz[0:32]
+			originalTxSenderPubkey := dataCipherBz[32:64]
```

Comment:
This feels very fragile to me if we change the schema of these messages. Do we have a canonical place in the code the parses these?